test/events.ts

```diff
@@ -0,0 +1,139 @@
+import {
+  getQuote,
+  getRouteTransactionData,
+  grantToken,
+  forkNetwork,
+  chain,
+  forkBlock,
+  waitFor,
+} from '../ts/utils';
+import { ethers, getNamedAccounts, deployments, network } from 'hardhat';
+import { SectorCrossVault } from '../typechain';
+
+// TODO orgainze these into a config file
+const USDC = '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48'; // USDC ETHEREUM
+
+const setupTest = deployments.createFixture(async () => {
+  await deployments.fixture(['XVault']);
+  const { deployer } = await getNamedAccounts();
+
+  // this contract will be grabbed form the deployments generated via the deployment fixture
+  // above, we can grab different contracts via the name we saved them as
+  const vault: SectorCrossVault = await ethers.getContract(
+    'SectorCrossVault-0',
+    deployer
+  );
+
+  // TODO: this can be moved to setup deployment
+  const whaleAddress = '0x0A59649758aa4d66E25f08Dd01271e891fe52199';
+
+  // Grant tokens to vault
+  await grantToken(vault.address, USDC, 5000, whaleAddress);
+  return vault;
+});
+
+describe('events', function () {
+  let vault: SectorCrossVault;
+  before(async () => {
+    // fork chain is set via .evn FORK_CHAIN param
+    await forkNetwork(chain, forkBlock[chain]);
+    vault = await setupTest();
+  });
+  it('should send events', async function () {
+    listenToEvents(vault);
```

Comment:
in reality we will be grabbing the event data from the tx receipt as opposed to listening for it.